nuxeo-runtime/nuxeo-runtime/src/test/java/org/nuxeo/runtime/TestDescriptorRegistry.java

```diff
@@ -0,0 +1,145 @@
+            TestDescriptor casted = (TestDescriptor) other;
+            TestDescriptor merged = new TestDescriptor();
+            merged.id = casted.id != null ? casted.id : id;
+            merged.name = casted.name != null ? casted.name : name;
+            merged.desc = casted.desc != null ? casted.desc : desc;
+            return merged;
+        }
+
+    }
+
+    private static final String EP = "ep1";
+
+    private static final String TARGET = "target1";
+
+    @Test
+    public void testGetDescriptors() {
+        DescriptorRegistry registry = new DescriptorRegistry();
+        registry.register(TARGET, EP, new TestDescriptor("id1", "name1", "desc1"));
+        registry.register(TARGET, EP, new TestDescriptor("id1", "name11", "desc1"));
+        registry.register(TARGET, EP, new TestDescriptor("id1", null, "desc11"));
+        registry.register(TARGET, EP, new TestDescriptor("id2", "name2", "desc2"));
+        registry.register(TARGET, EP, new TestDescriptor("id2", "name22", "desc2"));
+        registry.register(TARGET, EP, new TestDescriptor("id2", null, "desc22"));
+        List<TestDescriptor> descs = registry.getDescriptors(TARGET, EP);
+        Assert.assertEquals(2, descs.size());
+        assertValues(descs.get(0), "id1", "name11", "desc11");
+        assertValues(descs.get(1), "id2", "name22", "desc22");
+
+    }
+
+    @Test
+    public void testGetSingleDescriptor() {
+        DescriptorRegistry registry = new DescriptorRegistry();
+        registry.register(TARGET, EP, new TestDescriptor("id1", "name1", "desc1"));
+        assertValues((TestDescriptor) registry.getDescriptor(TARGET, EP, "id1"), "id1", "name1", "desc1");
+        registry.register(TARGET, EP, new TestDescriptor("id1", "name2", "desc1"));
+        assertValues((TestDescriptor) registry.getDescriptor(TARGET, EP, "id1"), "id1", "name2", "desc1");
+        registry.register(TARGET, EP, new TestDescriptor("id1", null, "desc2"));
+        assertValues((TestDescriptor) registry.getDescriptor(TARGET, EP, "id1"), "id1", "name2", "desc2");
+    }
+
+    @Test
+    public void testRemove() {
+        DescriptorRegistry registry = new DescriptorRegistry();
+        registry.register(TARGET, EP, new TestDescriptor("id0", "", ""));
+        assertValues(registry.getDescriptor(TARGET, EP, "id0"), "id0", "", "");
+        registry.register(TARGET, EP, new TestDescriptor("id0", "merged", "merged"));
+        assertValues(registry.getDescriptor(TARGET, EP, "id0"), "id0", "merged", "merged");
+        registry.register(TARGET, EP, new TestDescriptor("id0", "", "") {
+            @Override
+            public boolean remove() {
+                return true;
+            }
+        });
+        assertNull(registry.getDescriptor(TARGET, EP, "id0"));
+        registry.register(TARGET, EP, new TestDescriptor("id0", "final", null));
+        assertValues(registry.getDescriptor(TARGET, EP, "id0"), "id0", "final", null);
+    }
+
+    @Test
+    public void testSort() {
+
```

Comment:
the purpose of the test is not clear, it shows that descriptors order is preserved not ordered?

Reply:
This test has been removed it was useless since the order is not handled by the DescriptorRegistry anymore (for now at least)